frontend/src/tests/application-tests.js

```diff
@@ -1,24 +1,428 @@
+        });
+
+        it("Survey.js posting data is pre-filled based on prior applicant", async () => {
+            await switchToApplicantUser();
+
+            // Read survey.js posting data
+            let resp = await apiGET(
+                `/public/postings/${posting.url_token}`,
+                true
+            );
+            expect(resp).toHaveStatus("success");
+            checkPropTypes(surveyPropTypes, resp.payload);
+            const surveyPrefill = resp.payload["prefilled_data"];
+
+            for (const prefilledKey of [
+                "utorid",
+                "student_number",
+                "first_name",
+                "last_name",
+                "email",
+                "phone",
+            ]) {
+                // Check if the applicant prefill information is correct
+                expect(surveyPrefill[prefilledKey]).toEqual(
+                    applicant[prefilledKey]
+                );
+            }
+            await restoreDefaultUser();
+        });
+
         it.todo("Can submit survey.js data via the public postings route");
-        it.todo(
-            "When submitting survey.js data an applicant and application are automatically created they don't exist"
+
+        it.skip("Survey.js posting data is pre-filled based on prior application", async () => {
+            // Submit an application to the posting
+            const application = {
+                ...applicant,
+                program: "P",
+                program_start: "2021-06-21",
+                department: "cs",
+                previous_university_ta: false,
+                previous_department_ta: false,
+                previous_other_university_ta: false,
+                comments: "My comments",
+            };
+
+            let resp = await apiPOST(
+                `/public/postings/${posting.url_token}/submit`,
+                application,
+                true
+            );
+            expect(resp).toHaveStatus("success");
+
+            const applicationPrefillData = {
+                program: resp.payload.program,
+                department: resp.payload.department,
+                yip: resp.payload.yip,
+                previous_department_ta: resp.payload.previous_department_ta,
+                previous_university_ta: resp.payload.previous_university_ta,
+                previous_experience_summary:
+                    resp.payload.previous_experience_summary,
+                gpa: resp.payload.gpa,
+                comments: resp.payload.comments,
+            };
+
+            await switchToApplicantUser();
+
+            // Read survey.js posting data
+            resp = await apiGET(`/public/postings/${posting.url_token}`, true);
+            expect(resp).toHaveStatus("success");
+            checkPropTypes(surveyPropTypes, resp.payload);
+            expect(resp.payload["prefilled_data"]).toEqual(
+                expect.objectContaining(applicationPrefillData)
```

Comment:
This seems circular to me. The `applicationPrefillData` is created based on what comes from the backend. However, you never check that this data matches what you sent *to* the backend. All these values could be null, and you wouldn't notice...

Reply:
Here I am assuming that the surveyjs submission functions as expected since it is not a part of my test. This implies that `applicationPrefillData` cannot have `null` values except when we expect them to be `null`.